### ISSUE TITLE
Remove test infrastructure that is relevant to only Python 2.6 and earlier

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7826]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7826|
|Type|enhancement|
|Created|2015-03-18 10:50:26Z|
|Branch|https://github.com/twisted/twisted/tree/py26-tests-7826|

TestPyUnitResult.test_skip26 is only required on Python 2.6 because twisted.trial._synctest.SkipTest needs to make up for lack of skipping in 2.6 unittest

<details><summary>Searchable metadata</summary>

```
trac-id__7826 7826
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__EndOfLifePython2_6 EndOfLifePython2.6
branch__py26_tests_7826 py26-tests-7826
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__None None
time__1426675826047599 1426675826047599
changetime__1474198049187875 1474198049187875
version__None None
owner__Amber_Brown__HawkOwl___hawkowl_____ Amber Brown (HawkOwl) <hawkowl@...>
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

(In [45828]) Branching to py26-skiptest-7826.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

Builds are looking good. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Thanks for the cleanup.

--------------

It looks like this patch is also introducing a deprecation.... is it ok to merge only as a .misc?

Maybe we need a separate ticket for the deprecation.

----------

Some linters are failing.

```

twisted.trial.unittest.SynchronousTestCase._getSkipReason: invalid ref to unittest.SkipTest

twisted/trial/test/test_reporter.py:16: 'getmro' imported but unused

```

---------

This looks like a majore error... I run the builder again in builds/164 and I got the same error.


https://buildbot.twistedmatrix.com/builders/ubuntu12.04-py2.7/builds/160

```

/usr/lib/python2.7/dist-packages/zope/__init__.py:3: UserWarning: Module twisted was already imported from /home/buildslave/run/ubuntu12.04-py2.7/Twisted/twisted/__init__.py, but /home/buildslave/.local/lib/python2.7/site-packages is being added to sys.path
  import pkg_resources
Traceback (most recent call last):
  File "./bin/trial", line 21, in <module>
    from twisted.scripts.trial import run
  File "/home/buildslave/run/ubuntu12.04-py2.7/Twisted/twisted/scripts/trial.py", line 23, in <module>
    from twisted.trial import runner, itrial, reporter
  File "/home/buildslave/run/ubuntu12.04-py2.7/Twisted/twisted/trial/runner.py", line 36, in <module>
    from twisted.trial import util, unittest
  File "/home/buildslave/run/ubuntu12.04-py2.7/Twisted/twisted/trial/unittest.py", line 16, in <module>
    from twisted.trial._asyncrunner import (
  File "/home/buildslave/run/ubuntu12.04-py2.7/Twisted/twisted/trial/_asyncrunner.py", line 16, in <module>
    from twisted.trial import itrial, reporter
  File "/home/buildslave/run/ubuntu12.04-py2.7/Twisted/twisted/trial/reporter.py", line 35, in <module>
    if subunit.__version__ < (1, 0):
AttributeError: 'module' object has no attribute '__version__'

```


Please check the errors and resubmit.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc commented|
|-|-|

Since I have been looking through the codebase, I noticed some test infrastructure
relevant to Python <= 2.6, so it probably could be removed.

See:

https://github.com/twisted/twisted/pull/140

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Thanks for the patch.

I don't know what is the scope of your PR.

This ticket description talks about TestPyUnitResult.test_skip26 and about pyunit integration.

In your PR I don't see any changes related to pyunit.

I think that you will to update the description of your PR or create a new ticket in twisted as I don't know how your PR is related to this ticket.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_9"></a>|@rodrigc set owner to @adiroiban|
|-|-|

I have changed the title of this ticket to reflect the changes.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_10"></a>|@adiroiban removed owner|
|-|-|

I have not seen any comments from Amber. The initial ticket was about pyunit and Amber's branch was a work related to pyunit.

I have not see any comments about why the pyunit related work should be droped.

I have not see any comment about why your branch should not have a separate ticket.

Without any feedback from Amber, I feel like this ticket was hijacked :)

I am leaving this for other to review, as I don't know what is going on here.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_11"></a>|@hawkowl commented|
|-|-|

Pyunit was an earlier name for unittest, hence why I tried to remove it; it doesn't really make sense in the scope of the ticket, that was me just doing random work :)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_13"></a>|@hawkowl set owner to @hawkowl<br>@hawkowl set status to `closed`|
|-|-|

In changeset ab2b35051db3a18074685cc821ef485aa8f6d248
```
#!CommitTicketReference repository="" revision="ab2b35051db3a18074685cc821ef485aa8f6d248"
Merge py26-tests-7826: Remove some Python 2.6 test requirements

Author: rodrigc
Reviewer: hawkowl
Fixes: #7826
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_14"></a>|@hawkowl commented|
|-|-|

In changeset 43d19967e7ea4c350f31cd0392a47f1a4c4c242c
```
#!CommitTicketReference repository="" revision="43d19967e7ea4c350f31cd0392a47f1a4c4c242c"
Revert "Merge py26-tests-7826: Remove some Python 2.6 test requirements"

This reverts commit ab2b35051db3a18074685cc821ef485aa8f6d248, reversing
changes made to 84c674d2923aa3a57ff1f3c955591f669fdb81c9.

Reopens: #7826
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_15"></a>|@hawkowl commented|
|-|-|

In changeset 870658598c25546e6349e6fa27796d0122b56de8
```
#!CommitTicketReference repository="" revision="870658598c25546e6349e6fa27796d0122b56de8"
Merge py26-tests-7826: Remove some Python 2.6 test requirements

Author: rodrigc
Reviewer: hawkowl
Fixes: #7826
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_16"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
